eul/Views/Preference/PreferenceMenuViewView.swift

```diff
@@ -10,13 +10,157 @@ import SwiftUI
 
 extension Preference {
     struct PreferenceMenuViewView: View {
+        @State var updated = false
         @EnvironmentObject var preference: PreferenceStore
+        @State var dragging: EulComponent?
+        @State var frames: [CGRect] = .init(repeating: .zero, count: EulComponent.allCases.count)
+        @GestureState var offsetWidth: CGFloat = 0
+
+        func updateFrame(geometry: GeometryProxy, index: Int) -> some View {
+            if !preference.isActiveMenuComponentToggling {
+                DispatchQueue.main.async {
+                    self.frames[index] = geometry.frame(in: CoordinateSpace.named("ComponentsOrdering"))
+                }
+            }
+            return Color.clear
+        }
```

Comment:
it will be better to wrap the organizing logic (re-order/add/remove) into a component, rather than simply copy - paste. would you mind to take some time for that?